plugins/inputs/openweathermap/README.md

```diff
@@ -42,6 +47,8 @@ https://openweathermap.org/city/2643743
   - tags:
     - city_id
     - forecast
+    - condition_id
```

Comment:
> It's part of OWM's API and someone familiar with it would think it's missing if we don't include it

It's quite common for telegraf to not create a metric from everything provided by an api. Some things just don't make sense as time series data.

Reply:
Regarding the human readable vs id: telegraf has standardized enum "codes" to be stored as fields and the string representations to be stored as tags. One example is the `kube_inventory` plugin (`state` and `state_code`):
https://github.com/influxdata/telegraf/blob/817c9a69a94060aece50889c27e7c821a36a6530/plugins/inputs/kube_inventory/pod.go#L55-L63

>I would rather select ID=802

Very much agree. I'm not sure where/why you would do the other option. Both are still possible in either configuration from what I understand, but in influx, you can only `group by` tags.

Reply:
The influxdb schema design docs have some advice on choosing tags vs fields (https://docs.influxdata.com/influxdb/v1.7/concepts/schema_and_data_layout/). It says to make it a tag if it's commonly queried or used in a group by.  Only tags are indexed, so querying on fields is slow.

After reading those docs I think I'd like to leave id and main as tags and description as a field.  I imagine querying on and grouping by conditions will be common, so main and id should be tags.  If you want the general condition like snowy days, use main="snow".  If you want a specific condition like all days with freezing rain, use id="511".  I think description isn't going to be commonly queried because it's localized and too long, so it should be a field.

I think the difference between this and kube_inventory is that kube_inventory's state is short (one word), only has three values, and isn't localized.  That makes state easier to use than state_code, so it fits the tags criteria better.

I'd be fine leaving out icon but I don't think it's hurting anything.  It doesn't give you much over id and main for grouping or querying, so if it stays it should be a field.  It's only really useful if you build custom ui and need a weather conditions icon.  I'm leaning towards leaving it in and letting people filter it out if they don't want it.